### PR TITLE
add winston  logger

### PR DESCRIPTION
**Pull request by vasylkushnir**

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import taskRouter from './routes'
 import db from './lib/db'
 import config from './config'
 import errorHandler from './middlewares/errorHandler'
+import Logger from './lib/logger'
```
Comment (Ilchuk-Mihail) [inline src/app.ts:6]: import `logger`

File: src/controller/task.ts [hunk 2]
```diff
@@ -12,6 +13,10 @@ export default {
         importance,
         status
       })
+      Logger.info('Task created', {
+        id: newTask._id,
+        statusCode: res.statusCode
```
Comment (Ilchuk-Mihail) [inline src/controller/task.ts:18]: statusCode does not need to be logged (except errorHandler)

File: src/controller/task.ts [hunk 3]
```diff
@@ -32,6 +37,11 @@ export default {
           .limit(limit)
           .lean()
       ])
+      Logger.info('Total number of tasks are', {
```
Comment (Ilchuk-Mihail) [inline src/controller/task.ts:40]: Usually logs should be written on write operations POST/PUT/PATCH/DELETE but not GET. 
It's redundant to write logs on get operations.

File: src/middlewares/errorHandler.ts [hunk 4]
```diff
@@ -1,9 +1,14 @@
 import { Request, Response, NextFunction } from 'express'
 import BaseError from '../errors/BaseError'
+import Logger from '../lib/logger'
 
 export default function errorHandler (error: BaseError, req: Request, res: Response, next: NextFunction): void {
   const status = error.status || 500
   const message = error.message || 'Something went wrong '
+
+  Logger.error('Error Message:', { message })
+  Logger.error('Error Status Code', { statusCode: status })
```
Comment (Ilchuk-Mihail) [inline src/middlewares/errorHandler.ts:10]: Should be in one log message

**Review by Ilchuk-Mihail — CHANGES_REQUESTED** [review]

File: src/app.ts (hunk 1 above)
Comment (vasylkushnir) [reply src/app.ts]: fixed
e9acee3ca21fc4f0db8449487a6402632f34093e

File: src/controller/task.ts (hunk 3 above)
Comment (vasylkushnir) [reply src/controller/task.ts]: fixed e9acee3ca21fc4f0db8449487a6402632f34093e

File: src/middlewares/errorHandler.ts (hunk 4 above)
Comment (vasylkushnir) [reply src/middlewares/errorHandler.ts]: ашчув e9acee3ca21fc4f0db8449487a6402632f34093e

File: src/controller/task.ts (hunk 2 above)
Comment (vasylkushnir) [reply src/controller/task.ts]: fixed e9acee3ca21fc4f0db8449487a6402632f34093e

**Review by Ilchuk-Mihail — APPROVED** [review]

**Pull request merged**